internal/configs/static_evaluator.go

```diff
@@ -116,3 +116,12 @@ func (s StaticEvaluator) DecodeBlock(body hcl.Body, spec hcldec.Spec, ident Stat
 	diags = append(diags, valDiags...)
 	return val, diags
 }
+
+func (s StaticEvaluator) EvalContext(ident StaticIdentifier, refs []*addrs.Reference) (*hcl.EvalContext, hcl.Diagnostics) {
+	return s.EvalContextWithParent(nil, ident, refs)
+}
+
+func (s StaticEvaluator) EvalContextWithParent(parent *hcl.EvalContext, ident StaticIdentifier, refs []*addrs.Reference) (*hcl.EvalContext, hcl.Diagnostics) {
+	evalCtx, diags := s.scope(ident).EvalContextWithParent(parent, refs)
+	return evalCtx, diags.ToHCL()
+}
```

Comment:
Just a random thought, could probably just ignore and resolve.

I am now contemplating what's the best approach for `StaticEvaluator`, should we keep creating convenience functions like `DecodeExpression` and `DecodeBlock` or not?

Here, we could technically use a convenience method for `DecodeBody`, pretty similarly to the other methods here. The two exceptions being that 1. We do not want to use all references from the body. 2. We want to have the original eval context as parent

I guess we can just go ahead and see how `StaticEvaluator` starts shaping itself

Reply:
I don't have a reasoned answer here. I think the best we can do is to see where the `StaticEvaluator` goes and refactor it at some point.